This breaks ROCm build:

```
./tensorflow/stream_executor/gpu/asm_compiler.h:24:10: fatal error: tensorflow/stream_executor/cuda/cuda_driver.h: No such file or directory
 #include "tensorflow/stream_executor/cuda/cuda_driver.h"
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Note that asm_compiler is used for both ROCm and CUDA.

--- a/tensorflow/stream_executor/gpu/BUILD
+++ b/tensorflow/stream_executor/gpu/BUILD
@@ -243,6 +243,7 @@ cc_library(
         "//tensorflow/core:lib",
         "//tensorflow/core/platform:regexp",
         "//tensorflow/core/platform:cuda_libdevice_path",
+        "//tensorflow/stream_executor:kernel",
         "//tensorflow/stream_executor/lib",
         "//tensorflow/stream_executor/platform",
         "@com_google_absl//absl/strings",

--- a/tensorflow/stream_executor/gpu/asm_compiler.cc
+++ b/tensorflow/stream_executor/gpu/asm_compiler.cc
@@ -15,6 +15,10 @@ limitations under the License.
 
 #include "tensorflow/stream_executor/gpu/asm_compiler.h"
 
+#include <string>
+#include <utility>
+
+#include "absl/container/flat_hash_map.h"
 #include "absl/container/flat_hash_set.h"
 #include "absl/strings/str_format.h"
 #include "absl/strings/string_view.h"

--- a/tensorflow/stream_executor/gpu/asm_compiler.h
+++ b/tensorflow/stream_executor/gpu/asm_compiler.h
@@ -21,12 +21,14 @@ limitations under the License.
 #include "absl/container/flat_hash_map.h"
 #include "absl/types/span.h"
 #include "tensorflow/core/platform/mutex.h"
-#include "tensorflow/stream_executor/cuda/cuda_driver.h"
 #include "tensorflow/stream_executor/gpu/gpu_asm_opts.h"
 #include "tensorflow/stream_executor/kernel.h"
 #include "tensorflow/stream_executor/lib/statusor.h"
 #include "tensorflow/stream_executor/platform/port.h"
 #include "tensorflow/stream_executor/stream_executor_pimpl.h"
+#if GOOGLE_CUDA
+#include "tensorflow/stream_executor/cuda/cuda_driver.h"
+#endif  // GOOGLE_CUDA
 
 namespace stream_executor {
 namespace gpu {
@@ -85,6 +87,7 @@ port::StatusOr<std::vector<uint8>> BundleGpuAsm(
 port::StatusOr<std::vector<uint8>> LinkGpuAsm(
     gpu::GpuContext* context, std::vector<CubinOrPTXImage> images);
 
+#if GOOGLE_CUDA
 // Maintains a cache of pointers to loaded kernels
 template <typename... Args>
 port::StatusOr<std::shared_ptr<TypedKernel<Args...>>> LoadKernelOrGetPtr(
@@ -114,6 +117,7 @@ port::StatusOr<std::shared_ptr<TypedKernel<Args...>>> LoadKernelOrGetPtr(
   CHECK(it != kernel_ptr_cache.end());
   return it->second;
 }
+#endif  // GOOGLE_CUDA
 
 }  // namespace stream_executor
 

--- a/tensorflow/stream_executor/gpu/redzone_allocator.cc
+++ b/tensorflow/stream_executor/gpu/redzone_allocator.cc
@@ -322,11 +322,19 @@ port::StatusOr<RedzoneCheckStatus> RedzoneAllocator::CheckRedzones() const {
       executor->AllocateOwnedScalar<uint64>();
   stream_->ThenMemZero(out_param.ptr(), sizeof(uint64));
 
+#if GOOGLE_CUDA
   TF_ASSIGN_OR_RETURN(
       std::shared_ptr<ComparisonKernelT> loaded_kernel,
       (LoadKernelOrGetPtr<DeviceMemory<uint8>, uint8, uint64,
                           DeviceMemory<uint64>>(
           executor, "redzone_checker", redzone_checker_ptx, compiled_ptx)));
+#else
+  TF_ASSIGN_OR_RETURN(
+      std::unique_ptr<ComparisonKernelT> loaded_kernel,
+      (executor->CreateTypedKernel<DeviceMemory<uint8>, uint8, uint64,
+                                   DeviceMemory<uint64>>(
+          "redzone_checker", redzone_checker_ptx, compiled_ptx)));
+#endif  // GOOGLE_CUDA
 
   for (const auto& buf_and_size : allocated_buffers_) {
     TF_ASSIGN_OR_RETURN(

